### PR TITLE
Refactor configurable testing code

### PR DESCRIPTION
**Pull request by farshidtz**
- [ ] Remove temp branch for CI tests

### CODE REVIEW

File: test/utils/common.go [hunk 1]
```diff
@@ -4,33 +4,28 @@ import (
...
+	if params.TestMixedConfig {
+		SetMixedConfig(t, params.Snap, params.App, params.DefaultServicePort)
 	}
 
+	// networking tests
 	if len(params.TestOpenPorts) > 0 {
 		for _, port := range params.TestOpenPorts {
 			WaitServiceOnline(t, 60, port)
 		}
 	}
```
Comment (MonicaisHer) [inline test/utils/common.go:27]: Is this subtest redundant with [utils.WaitServiceOnline](https://github.com/canonical/edgex-snap-testing/blob/7e2db2ca2ca71bfe66e66f2a304e0f6d6b67f0bb/test/suites/edgexfoundry/main_test.go#L55)?

Comment (MonicaisHer) [inline test/utils/common.go:27]: Is this WaitServiceOnline() redundant with [utils.WaitPlatformOnline(nil)](https://github.com/canonical/edgex-snap-testing/blob/7e2db2ca2ca71bfe66e66f2a304e0f6d6b67f0bb/test/suites/edgexfoundry/main_test.go#L37)? And WaitServiceOnline() didn't test 8000, 5432 and 6379.

Comment (farshidtz) [reply test/utils/common.go:27]: I think the wait in the main is redundant. We should increase the interval delay and avoid relying on the startup order. See https://github.com/canonical/edgex-snap-testing/issues/30

File: test/suites/edgexfoundry/main_test.go [hunk 2]
```diff
@@ -52,23 +52,35 @@ func TestMain(m *testing.M) {
...
 			TestAppConfig:      true,
 			TestGlobalConfig:   true,
 			TestMixedConfig:    utils.FullConfigTest,
-			DefaultServicePort: []string{supportSchedulerServicePort},
+			DefaultServicePort: supportSchedulerServicePort,
 		},
-		TestNetworking: utils.TestNetworking{
-			TestOpenPorts:        utils.PlatformPorts,
+		NetworkingTests: utils.NetworkingTests{
+			TestOpenPorts:        platformPorts,
```
Comment (MonicaisHer) [inline test/suites/edgexfoundry/main_test.go:79]: WaitServiceOnline(platformPorts) didn't test 8000, 5432 and 6379.
It needs utils.PlatformPorts here.

**Comment by farshidtz** [comment]
@MonicaisHer I'm done with refactoring and a bunch of minor fixes. It is ready for re-review.

File: test/utils/config.go [hunk 3]
```diff
@@ -2,8 +2,52 @@ package utils
...
+		if conf.TestGlobalConfig {
+			SetGlobalConfig(t, snapName, conf.App, conf.DefaultPort)
+		}
+		if conf.TestMixedGlobalAppConfig {
+			SetMixedConfig(t, snapName, conf.App, conf.DefaultPort)
+		}
+	})
+}
+
+// TODO change to TestChangePortLegacyEnv
```
Comment (MonicaisHer) [inline test/utils/config.go:50]: Why is the change from `SetEnvConfig` to `TestChangePortLegacyEnv` needed?

Comment (farshidtz) [reply test/utils/config.go:50]: This is consistent with the other function name changes, to make it clear that:
1) This is executing a test (t.Run)
2) This is testing the changing of port

**Review by MonicaisHer — APPROVED** [review]
@farshidtz Thanks a lot for the changes.

**Pull request merged**